src/pages/blocks/centre.tsx

```diff
@@ -0,0 +1,327 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Box,
+  Typography,
+  TextField,
+  Button,
+  InputAdornment,
+} from '@mui/material';
+import Header from '../../components/Header';
+import { useTheme } from '@mui/material/styles';
+import { useTranslation } from 'next-i18next';
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
+import Search from '@mui/icons-material/Search';
+import { Apartment, ArrowDropDown, KeyboardArrowRight, SmartDisplayOutlined } from '@mui/icons-material';
+import AddIcon from '@mui/icons-material/Add';
+import FilterModalCenter from './components/FilterModalCenter';
+
+interface BlocksProps {
+  // Define any props if needed
+}
+
+const Blocks: React.FC<BlocksProps> = () => {
+  const { t } = useTranslation();
+  const theme = useTheme<any>();
+  const [districts, setDistrict] = useState('');
+  const [blocks, setBlocks] = useState('');
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const [filterModalOpen, setFilterModalOpen] = useState(false); // State for filter modal
+  const [searchInput, setSearchInput] = useState(''); // State for search input
+  const [centers, setCenters] = useState([
+    'Bhivapur',
+    'Kamptee',
+    'Katol',
+    'Kuhi',
+    'Mauda',
+    'Savner',
+    'ABC',
+    'DEF',
+    'XYZ',
+  ]); // Initial centers
+  const [filteredCenters, setFilteredCenters] = useState(centers); // State for filtered centers
+  const [selectedCenters, setSelectedCenters] = useState<string[]>([]); // State for selected centers
+  const [sortOrder, setSortOrder] = useState('asc'); // State for sort order
+  const [centerType, setCenterType] = useState(''); // State for center type
+
+  const handleFilterModalOpen = () => setFilterModalOpen(true); // Function to open filter modal
+  const handleFilterModalClose = () => setFilterModalOpen(false); // Function to close filter modal
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchInput(event.target.value);
+    filterCenters(event.target.value);
+  };
+
+  const filterCenters = (searchTerm: string) => {
+    setFilteredCenters(
+      centers?.filter((center) =>
+        center.toLowerCase().includes(searchTerm.toLowerCase())
+      )
+    );
+  };
+
+  const applyFilters = () => {
+    // Apply sorting and filtering based on selected options
+    let updatedCenters = centers;
+
+    // Apply sorting
+    if (sortOrder === 'asc') {
+      updatedCenters = updatedCenters?.sort();
+    } else if (sortOrder === 'desc') {
+      updatedCenters = updatedCenters?.sort().reverse();
+    }
+
+    // Apply center type filter
+    if (centerType === 'regular') {
+      updatedCenters = updatedCenters?.filter((center) =>
+        ['Bhivapur', 'Kamptee', 'Katol'].includes(center)
```

Comment:
@Aar-if dont hard code the center, instead add new key in the center object as `remote: true` and use same here

Reply:
Yes